src/transport.js

```diff
@@ -67,14 +68,19 @@ module.exports = function (swarm) {
               log('dial canceled: %s', multiaddr.toString())
               // clean up already done dials
               if (conn) {
-                conn.close()
+                pull(
+                  pull.empty(),
+                  conn
+                )
+
+                // conn.close()
               }
               return cb()
             }
 
             // one is enough
             log('dial success: %s', multiaddr.toString())
-            q.kill()
+            q.kill() // why kill here?
```

Comment:
@dignifiedquire why kill the queue? You are queue by transport, if I issue two TCP dials to two different peers, the success of one will kill the other.